greynoise/actions/lib/base.py

```diff
@@ -0,0 +1,23 @@
+from netaddr import IPAddress
+from st2common.runners.base_action import Action
+
+from greynoise import GreyNoise
+
+
+class GreyNoiseBaseAction(Action):
+    def __init__(self, config):
+        super(GreyNoiseBaseAction, self).__init__(config=config)
+
+        gn_api_key = self.config.get('gn_api_key', None)
+        self.instance = GreyNoise(api_key=gn_api_key,
+                                  integration_name="greynoise-stackstorm-v1.0.0")
```

Comment:
@blag : I agree, I'm not thrilled with hardcoding this value, but we use this for tracking metrics on our side, so desire to keep a version here.  Is there a way that I can dynamically pull this is from the client when it is initialized?  I'll at least move it out to a variable for now, so it is more obvious, but if there is a way to pull it from the pack.yaml, that would be perfect.

Reply:
I'm not aware of a way to pull it from the `pack.yaml`. 😕

I understand the "need" to track for metrics, but I'm not so sure that it's going to have as much value as you'd like.

Consider the following: we release this pack as v1.0.0, and then at some point the pack gets revved to v1.0.1, and then v1.1.0, and then 1.2.0. But since this pack is a bit of a unicorn in this regard, the version here is not updated, so all of those pack revisions will still use the `greynoise-stackstorm-v1.0.0` integration name.

In that case, you will have _worse_ data, because it will be lying to you. People who are using v1.0.1, v1.1.0, and v1.2.0 will all show up in your metrics as using the v1.0.0 of this pack.

Instead, I would just use the integration name `greynoise-stackstorm`. It doesn't include a version name, sure, but at least your metrics are no longer lying to you.

Alternatively, you could add an immutable key to the pack config schema <sup>1</sup>, with the default value set to the (current) pack value. Pull the version from the pack config in your code, and drop a YAML comment next to or on the same line as the version in `pack.yaml` to please also update the version in the pack config schema:

`pack.yaml`:
```yaml
...
version: 1.0.0  # Please also update the version in the config.schema.yaml!
...
```

That is by no means bulletproof, but if you're lucky your metrics will be slightly more accurate.

I did just have one idea, let me bounce this off of you: GreyNoise can donate money, and we would provide you with the raw metrics from GitHub. We would have to make sure that we can get the metrics from GitHub with the detail that would be valuable to you, but that might be a win-win. StackStorm has been searching for ways to get more community contributions, and that might be a good way. Now, of course, I/we would have to discuss this with the TSC and get approval for this, so I can't promise anything here and now.

As of now, to my knowledge, you are the only person to have requested this feature of StackStorm. If this is a common use-case for integration packs, then I think we would be amenable to a PR to ST2 that made it easier to access values from `pack.yaml` files.

<sup>1</sup> However, I would _not_ include the key in the config example, because you don't want to allow users to override it anyway. And you don't want to hint at it, because if the value is marked as immutable in the config schema, then they won't be able to override it anyway.

Reply:
Per the suggestion from @nmaludy : I was able to get this by doing a read/regex from the pack.yaml file, so this is now dynamically pulled whenever it is updated.  Thanks for the suggestions here.